ext/signals.c

```diff
@@ -102,6 +107,138 @@ static bool ddtrace_crashtracker_check_result(ddog_VoidResult result, const char
     return true;
 }
 
+#if PHP_VERSION_ID >= 80000
+static zend_never_inline ZEND_COLD void ddtrace_crasht_failed_tag_push(
+    ddog_Error *err,
+    ddog_CharSlice key
+) {
+    ddog_CharSlice msg = ddog_Error_message(err);
+    LOG(DEBUG,
+        "Failed to push tag \"%.*s\": %.*s",
+        (int) key.len, key.ptr,
+        (int) msg.len, msg.ptr);
+    ddog_Error_drop(err);
+}
+
+ // This doesn't increase the refcount of the returned string (no cleanup).
+static zend_always_inline zend_string *ddtrace_crasht_ini_get_value(ddog_CharSlice ini) {
+    zend_string *ini_zstr = zend_string_init(ini.ptr, ini.len, true);
+    zend_string *value = zend_ini_get_value(ini_zstr);
+    zend_string_release(ini_zstr);
+    return value;
+}
+
+// Pass in a key like "php.opcache.enable" and "php." will get stripped off,
+// and that's what the INI setting will be.
+static void ddtrace_crasht_add_opcache_tag(
+    ddog_Vec_Tag *tags,
+    ddog_CharSlice key
+) {
+    const ddog_CharSlice PREFIX = DDOG_CHARSLICE_C("php.");
+    ZEND_ASSERT(key.len > PREFIX.len && memcmp(key.ptr, PREFIX.ptr, PREFIX.len) == 0);
+
+    ddog_CharSlice ini = {
+        .ptr = key.ptr + PREFIX.len,
+        .len = key.len - PREFIX.len,
+    };
+    zend_string *value = ddtrace_crasht_ini_get_value(ini);
```

Comment:
```suggestion
    const char *value = zend_ini_string(key.ptr + PREFIX.len, key.len - PREFIX.len, 0);
```
(using zend_ini_str and polyfilling it in compatibility.h would be even better)

Reply:
Hmm, zend_ini_string returns a `char*`. Wish there was an API for passing a ptr + len for the INI but getting back a `zend_string *`. In any case, yeah, we can avoid the allocation.

Reply:
I've polyfilled it for older versions. However, note that I'm not using it on PHP <8 so... it may not compile. It looks like it should, based on looking at API functions on heap.space.

Reply:
Yes, that polyfill will work perfectly on PHP 7.